### PR TITLE
Cmr 4229

### PR DESCRIPTION
**Pull request by laurenfrederick**
Community usage metrics were not being found for collections where the version format in the collection did not match the version in the EMS metrics file.

Move parse-version-id to a common file to be used by indexing and community usage. Parse the version id when comparing community usage.

I've added binning the keyword score and tests for that, but currently have it disabled. I need to get some more tests that shows that working and also test performance of the script in WL.

I've also changed some of the int tests to use umm-spec rather than umm-lib.

### CODE REVIEW

File: system-int-test/test/cmr/system_int_test/search/collection_relevancy/collection_relevancy.clj [hunk 1]
```diff
@@ -83,4 +90,29 @@
...
+     ;; If the scores change here, this test will either fail or become useless and
+     ;; will have to be rewritten
+     (is (= [0.65 0.5 0.5]
+            (map :score (:refs (search/find-refs :collection {:keyword "Usage"})))))
+
+     (dev-sys-util/eval-in-dev-sys `(query-to-elastic/set-sort-bin-keyword-scores! true))
+
+     (are3 [size expected-collections]
+       (do
+         (def bin-size size) ; Setting the config will fail without this
```
Comment (chris-durbin) [inline system-int-test/test/cmr/system_int_test/search/collection_relevancy/collection_relevancy.clj]: This doesn't make sense to me. Why doesn't ```(dev-sys-util/eval-in-dev-sys `(query-to-elastic/set-keyword-score-bin-size! size))``` work?

**Comment by chris-durbin** [comment]
Note you will have to pull master into your branch and resolve conflicts before merging.

File: indexer-app/src/cmr/indexer/data/concepts/collection/collection_util.clj [hunk 2]
```diff
@@ -0,0 +1,12 @@
+(ns cmr.indexer.data.concepts.collection.collection-util
+  "Contains util functions for collection indexing")
+
+(defn parse-version-id
+  "Safely parse the version-id to an integer and then return as a string. This
+  is so that collections with the same short name and differently formatted
+  versions (i.e. 1 vs. 002) can be more accurately sorted. If the version
```
Comment (ygliuvt) [inline indexer-app/src/cmr/indexer/data/concepts/collection/collection_util.clj]: Change to (i.e. 1 vs. 001)

Comment (laurenfrederick) [inline indexer-app/src/cmr/indexer/data/concepts/collection/collection_util.clj]: Done.

**Pull request merged**